core/models/DbModel.php

```diff
@@ -47,6 +74,46 @@ public static function findOne($where)
         return $statement->fetchObject(static::class);
     }
 
+    public static function getAll()
+    {
+        $tableName = static::tableName();
+        $statement = self::prepare("SELECT * FROM $tableName");
+        $statement->execute();
+
+        return $statement->fetchAll(\PDO::FETCH_ASSOC);
+    }
+
+    public static function getByID($_id)
+    {
+        $tableName = static::tableName();
+        $attributes = array_keys($_id);
+        $sql = implode(" OR ", array_map(fn ($attr) => "$attr = :$attr", $attributes));
```

Comment:
An ID array item consists of a ["key" => "value]. Therefore, `getByID` will receive array like:
```php
["id" => 1, "cart_id" => 2] 
```
Then the query is formed like it should be. However, I just realized that the implode function will ignore similar keys. For example:
```php
["id" => 1, "id" => 2] 
```
The query will only have the first item's key and value. Meaning there is no OR. I think this could be confusing but given the current use case, it works. I'll have to look for better approach.

Reply:
Yes, that was the scenario I was talking about. The way you would typically use this is to get a few items with a specific ID. It's rare to use an OR on different column constraints.